crates/subspace-farmer/src/farmer_piece_getter.rs

```diff
@@ -458,6 +493,36 @@ where
 
         piece_getter.get_piece(piece_index).await
     }
+
+    async fn get_pieces<'a, PieceIndices>(
+        &'a self,
+        piece_indices: PieceIndices,
+    ) -> anyhow::Result<
+        Box<dyn Stream<Item = (PieceIndex, anyhow::Result<Option<Piece>>)> + Send + Unpin + 'a>,
+    >
+    where
+        PieceIndices: IntoIterator<Item = PieceIndex, IntoIter: Send> + Send + 'a,
+    {
+        let Some(piece_getter) = self.upgrade() else {
+            debug!("Farmer piece getter upgrade didn't succeed");
+            return Ok(Box::new(stream::iter(
```

Comment:
Why it's not an error?

Reply:
This most likely means that the whole system is shutting down, no need to print an error about it.